disnake/guild_scheduled_event.py

```diff
@@ -0,0 +1,372 @@
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Any, Dict, List, Optional, TYPE_CHECKING, Union
+from .enums import (
+    GuildScheduledEventEntityType,
+    GuildScheduledEventStatus,
+    StagePrivacyLevel,
+    try_enum,
+)
+from .user import User
+from .member import Member
+from .mixins import Hashable
+from .utils import cached_slot_property, parse_time, _get_as_snowflake, MISSING
+
+if TYPE_CHECKING:
+    from .abc import GuildChannel
+    from .guild import Guild
+    from .state import ConnectionState
+
+
+__all__ = ("GuildScheduledEventMetadata", "GuildScheduledEvent")
+
+
+class GuildScheduledEventMetadata:
+    """
+    Represents guild event entity metadata.
+
+    .. versionadded:: 2.3
+
+    Attributes
+    ----------
+    location: Optional[:class:`str`]
+        Location of the event
+    """
+
+    __slots__ = ("location",)
+
+    def __init__(self, *, location: str = None):
+        self.location: Optional[str] = location
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {"location": self.location}
+
+    @classmethod
+    def from_dict(cls, data):
+        return GuildScheduledEventMetadata(location=data.get("location"))
+
+
+class GuildScheduledEvent(Hashable):
+    """
+    Represents guild scheduled events.
+
+    .. versionadded:: 2.3
+
+    .. container:: operations
+
+        .. describe:: x == y
+
+            Checks if two scheduled events are equal.
+
+        .. describe:: x != y
+
+            Checks if two scheduled events are not equal.
+
+        .. describe:: hash(x)
+
+            Returns the scheduled event's hash.
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The id of the scheduled event.
+    guild_id: :class:`int`
+        The guild id which the scheduled event belongs to.
+    channel_id: Optional[:class:`int`]
+        The channel id in which the scheduled event will be hosted.
+        This field is ``None`` if :attr:`entity_type` is :class:`GuildScheduledEventEntityType.external`
```

Comment:
I think that when referring to the id, it should be in upper case (Basically to follow the same syntax in the documentation)